src/coreclr/jit/gentree.cpp

```diff
@@ -20897,18 +20951,18 @@ GenTree* Compiler::gtNewSimdCmpOpAllNode(
 #if defined(TARGET_XARCH)
         case GT_EQ:
         {
-            if (simdSize == 64)
-            {
-                assert(IsBaselineVector512IsaSupportedDebugOnly());
-                intrinsic = NI_Vector512_op_Equality;
-            }
-            else if (simdSize == 32)
+            if (simdSize == 32)
```

Comment:
just curious, why we decided to check for `32` , then `64` and then (implicit) `16` or less? Is it because `Vector256` is more common and should hit that condition first from TP perspective?

Reply:
Yep. Initially when we had throughput issues, this was one of the things we tried. Making check for 32 the first(being the most common case)